share/shwap/range_namespace_data.go

```diff
@@ -0,0 +1,160 @@
+package shwap
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/celestiaorg/celestia-app/v2/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/v2/pkg/wrapper"
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/shwap/pb"
+	"github.com/celestiaorg/nmt"
+)
+
+// RangedNamespaceDataFromShares builds a range of namespaced data for the given coordinates.
+// rawShares is a list of shares relative to the data square needed to build the range;
+// namespace is the target namespace for the built range;
+// rowIndex is the starting index of the list relative to the data square;
+// from is the share index of the first share of the range;
+// to is an EXCLUSIVE index of the last share of the range.
+func RangedNamespaceDataFromShares(
+	rawShares [][]share.Share,
+	namespace share.Namespace,
+	rowIndex, from, to int,
+) (NamespaceData, error) {
```

Comment:
Why not make it return RangeNamespaceData? To avoid handling proofs only in it?

Reply:
Yes and to avoid passing SampleID as it's a part of `RangeNamespaceData`